matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/room/model/livelocation/LiveLocationBeaconContent.kt

```diff
@@ -18,11 +18,25 @@ package org.matrix.android.sdk.api.session.room.model.livelocation
 
 import com.squareup.moshi.Json
 import com.squareup.moshi.JsonClass
+import org.matrix.android.sdk.api.session.events.model.Content
 import org.matrix.android.sdk.api.session.room.model.message.LocationAsset
 import org.matrix.android.sdk.api.session.room.model.message.LocationAssetType
+import org.matrix.android.sdk.api.session.room.model.message.MessageContent
+import org.matrix.android.sdk.api.session.room.model.message.MessageType
+import org.matrix.android.sdk.api.session.room.model.relation.RelationDefaultContent
 
 @JsonClass(generateAdapter = true)
 data class LiveLocationBeaconContent(
```

Comment:
The only way I found to be able to show the content as a message in the timeline is to inherit from `MessageContent` even it does not correspond to a message event but a state event. Is there a better way to implement it?

Reply:
it looks like we use this pattern for other timeline types, eg stickers, polls, verification. Would lean towards this being ok, although will delegate to @ganfra